Initial flask setup

UTkzhang opened the pull request:
Initial set up for flask backend, including models, sqlite db, api with openapi documentation (REST), mailing service, and docker.

UTkzhang committed:
flask backend infrastructure initial setup commit

UTkzhang committed:
typo fix on initializer

UTkzhang committed:
test

UTkzhang committed:
test2

UTkzhang committed:
fixed email.ini and committed template

UTkzhang committed:
server up

UTkzhang committed:
email tested

UTkzhang committed:
email tested

UTkzhang committed:
docker for flask tested

UTkzhang committed:
Create README.md

UTkzhang committed:
updated gitignore

eanile commented on PlateUp-flask/Dockerfile:
```diff
@@ -0,0 +1,6 @@
+COPY . /app
+WORKDIR /app
+RUN pip install -r requirements.txt
+ENTRYPOINT ["python"]
```
Should we specify python3 just in case?

UTkzhang replied:
sure

UTkzhang committed:
Merge branch 'initial-flask-setup' of https://github.com/ECE444-2020Fall/project1-plateup-group5-pyromaniacs into initial-flask-setup

UTkzhang committed:
fixed ds store issues

eanile commented on .gitignore:
```diff
@@ -4,3 +4,6 @@ PlateUp/.expo-shared
 PlateUp/node_modules
+
+# Flask
+PlateUp-flask/email_config.ini
```
Is there a reason this is ignored?

UTkzhang replied:
email password/user, should be configured locally

UTkzhang committed:
fixed status code on mail

UTkzhang committed:
fixed status code on conflict

UTkzhang committed:
changed python to python3 Dockerfile

UTkzhang committed:
Update README.md

UTkzhang committed:
react folder rename

UTkzhang committed:
Merge branch 'initial-flask-setup' of https://github.com/ECE444-2020Fall/project1-plateup-group5-pyromaniacs into initial-flask-setup

eanile commented on PlateUp-flask/emailservice.py:
```diff
@@ -0,0 +1,29 @@
+import configparser
+import smtplib
+from email.mime.text import MIMEText
```
nit: try and keep the format of the imports the same in each py file. Example: all "import ..." statements before "from __ import ..." statements, and keep the statements within each group in alphabetical order.

UTkzhang replied:
fixed

UTkzhang committed:
fixed imports ordering

eanile commented on PlateUp-flask/run.py:
```diff
@@ -0,0 +1,203 @@
+    @loginR.doc(description="Logging current user out.")
+    def delete(self):
+        logout_user(current_user)
+        return Response("Login successful. User %s"%current_user.id, status=200)
```
Login successful -> Logout successful

eanile replied:
Done.

eanile commented on PlateUp-flask/run.py:
```diff
@@ -0,0 +1,203 @@
+@mailR.route('')
+class Mail(Resource):
+    # @login_required
+    @mailR.doc(description="Sends a welcome email to user with their client ID and default password information (Development Email Only).")
```
nit: line too long

eanile commented on PlateUp-flask/run.py:
```diff
@@ -0,0 +1,203 @@
+def updateRecipes():
+    print("updating recipes...")
+    # do some process
+    time.sleep(10)
```
I'm assuming this is temporary?

eanile commented on PlateUp-flask/models.py:
```diff
@@ -1,9 +1,10 @@
+import json
```
Still not in alpha order. Also other files have this issue. Not a huge thing, just something I noticed.

eanile replied:
Fixed.

eanile committed:
Several formatting changes, along with some logic changes:
1. UUIDs for the user are converted to strings
2. Fixed bare except in emailservice
3. GET for UserAPI returns results, not results.data, as it isn't a thing
4. Changed id field for LoginAPI to string from int
5. Removed unnecessary argument to logout_user
6. Get user ID of user logging out before they logout
7. user_loader passes the UID directly instead of converting it to int

eanile commented:
NOTE: also please change class names of APIs to *API. i.e. Login -> LoginAPI.
